### PR TITLE
feat: enforce that account_ids are valid in FunctionCallPermission

### PR DESCRIPTION
**Pull request by matklad**
closes https://nearinc.atlassian.net/browse/CPR-135

### CODE REVIEW

File: runtime/runtime/src/verifier.rs [hunk 1]
```diff
@@ -417,6 +430,30 @@ fn validate_add_key_action(
     Ok(())
 }
 
+fn truncate_string(s: &str, limit: usize) -> String {
```
Comment (mm-near) [inline runtime/runtime/src/verifier.rs:433]: I'm surprised that something like this is not present in default Rust string library..

(or is it there, but we need a custom method for a reason ?)

File: runtime/runtime/src/verifier.rs [hunk 2]
```diff
@@ -417,6 +430,30 @@ fn validate_add_key_action(
...
+    unreachable!()
+}
+
+#[test]
+fn test_truncate_string() {
+    fn check(input: &str, limit: usize, want: &str) {
+        let got = truncate_string(input, limit);
+        assert_eq!(got, want)
+    }
+    check("hello", 0, "");
```
Comment (mm-near) [inline runtime/runtime/src/verifier.rs:448]: can you also check:

check("", 10, "")

File: integration-tests/src/tests/client/process_blocks.rs [hunk 3]
```diff
@@ -2670,6 +2670,107 @@ fn test_wasmer2_upgrade() {
...
+        let mut block = env.clients[0].produce_block(tip.height + 1).unwrap().unwrap();
+        set_block_protocol_version(&mut block, block_producer, new_protocol_version);
+        let (_, res) = env.clients[0].process_block(block.clone().into(), Provenance::NONE);
+        assert!(res.is_ok());
+        for i in 0..6 {
+            env.produce_block(0, tip.height + i + 2);
+        }
+    }
+
+    // Re-run the transaction.
```
Comment (mm-near) [inline integration-tests/src/tests/client/process_blocks.rs:2759]: and comment that we expect this transaction to fail, as this is the protocol version that enforces the account ids.

**Review by mm-near — APPROVED** [review]

File: runtime/runtime/src/verifier.rs (hunk 1 above)
Comment (matklad) [reply runtime/runtime/src/verifier.rs:433]: I am not surprised -- doing this properly is a hard and domain-specific: you probably don't want to truncate 🏳️‍🌈 to 🏳️ and for that you need to pull substantial amount of unicode in. And then you can truncate to the number of visual characters, or to the number of fixed-width symbols, or to the number of bytes.

**Pull request merged**